### PR TITLE
utility to convert to pyarrow table with vectors

### PR DESCRIPTION
**Pull request by changhiskhan**
addresses #563

### CODE REVIEW

File: python/python/lance/vector.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+"""Embedding vector utilities"""
+from typing import Union, Optional
+
+import numpy as np
+import pyarrow as pa
+
+#  Copyright 2023 Lance Developers
```
Comment (eddyxu) [inline python/python/lance/vector.py:7]: License above imports.

File: python/python/tests/test_vector.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+
+import numpy as np
+import pyarrow as pa
+import pytest
+
+from lance.vector import to_vectors
+
+#  Copyright 2023 Lance Developers
```
Comment (eddyxu) [inline python/python/tests/test_vector.py:8]: license upbove imports.

File: python/python/lance/vector.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    for v in values:
+        if ndim is None:
+            ndim = len(v)
+        else:
+            if ndim != len(v):
+                raise ValueError(f"Expected {ndim} dimensions but got {len(v)} for {v}")
+    return ndim
+
+
+def to_vectors(data: Union[dict, list, np.ndarray],
```
Comment (eddyxu) [inline python/python/lance/vector.py:38]: vector_to_table()?

File: python/python/lance/vector.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+               names: Optional[Union[str, list[str]]] = None,
+               ndim: Optional[int] = None,
+               check_ndim: bool = True) -> pa.Table:
```
Comment (eddyxu) [inline python/python/lance/vector.py:41]: do you need check dyppe = f32?

File: python/python/lance/vector.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    """
+    Create a pyarrow Table containing vectors.
+    Vectors are created as FixedSizeListArray's in pyarrow with Float32 values.
+
```
Comment (eddyxu) [inline python/python/lance/vector.py:45]: Maybe add an example section here? The usage is not clear.

File: python/python/lance/vector.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    Parameters
+    ----------
+    data: dict, list, or np.ndarray
+        If dict, the keys are added as "id" column
+        If list, then each element is assumed to be a vector
```
Comment (eddyxu) [inline python/python/lance/vector.py:50]: This only support 1-d vector, right?

File: python/python/lance/vector.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        If ndarray, then each row is assumed to be a vector
+    names: str or list, optional
+        If data is dict, then names should be a list of 2 str; default ["id", "vector"]
+        If data is list or ndarray, then names should be str; default "vector"
+    ndim: int, optional
+        Number of dimensions of the vectors. Inferred if omitted.
+    check_ndim: bool, default True
```
Comment (eddyxu) [inline python/python/lance/vector.py:57]: Add a `Return` section?

File: python/python/lance/vector.py (hunk 4 above)
Comment (changhiskhan) [reply python/python/lance/vector.py:41]: i just force it to f32

File: python/python/lance/vector.py (hunk 6 above)
Comment (changhiskhan) [reply python/python/lance/vector.py:50]: yes, list of 1-d vectors

File: python/python/lance/vector.py (hunk 1 above)
Comment (changhiskhan) [reply python/python/lance/vector.py:7]: fixed

File: python/python/tests/test_vector.py (hunk 2 above)
Comment (changhiskhan) [reply python/python/tests/test_vector.py:8]: fixed

File: python/python/lance/vector.py (hunk 3 above)
Comment (changhiskhan) [reply python/python/lance/vector.py]: changed to `vec_to_table`

File: python/python/lance/vector.py (hunk 7 above)
Comment (changhiskhan) [reply python/python/lance/vector.py:78]: added

File: python/python/lance/vector.py (hunk 5 above)
Comment (changhiskhan) [reply python/python/lance/vector.py:48]: added

File: python/python/lance/vector.py (hunk 1 above)
Comment (eddyxu) [reply python/python/lance/vector.py:3]: Docstring is still on top of license. The file should start with license

**Review by eddyxu — APPROVED** [review]
One more license fix. rest is good

**Pull request merged**